WooCommerce/Classes/ViewRelated/Products/Media/WordPressMediaLibraryPickerDataSource.swift

```diff
@@ -0,0 +1,189 @@
+        onDataChange = nil
+    }
+
+    func unregisterGroupChangeObserver(_ blockKey: NSObjectProtocol) {
+        // The group never changes
+    }
+
+    func loadData(with options: WPMediaLoadOptions, success successBlock: WPMediaSuccessBlock?, failure failureBlock: WPMediaFailureBlock? = nil) {
+        syncingCoordinator.resetInternalState()
+        retrieveMedia(pageNumber: syncingCoordinator.pageFirstIndex, pageSize: Constants.numberOfItemsPerPage) { [weak self] (mediaItems, error) in
+            guard error == nil else {
+                failureBlock?(error)
+                return
+            }
+            self?.mediaItems = mediaItems
+            successBlock?()
+        }
+    }
+
+    func mediaTypeFilter() -> WPMediaType {
+        return .image
+    }
+
+    func ascendingOrdering() -> Bool {
+        return true
+    }
+
+    func add(_ image: UIImage, metadata: [AnyHashable: Any]?, completionBlock: WPMediaAddedBlock? = nil) {}
+
+    func addVideo(from url: URL, completionBlock: WPMediaAddedBlock? = nil) {}
+
+    func setMediaTypeFilter(_ filter: WPMediaType) {}
+
+    func setAscendingOrdering(_ ascending: Bool) {}
+
+    func setSelectedGroup(_ group: WPMediaGroup) {}
+}
+
+extension WordPressMediaLibraryPickerDataSource: SyncingCoordinatorDelegate {
+    func sync(pageNumber: Int, pageSize: Int, reason: String?, onCompletion: ((Bool) -> Void)?) {
+        retrieveMedia(pageNumber: pageNumber, pageSize: pageSize) { [weak self] (mediaItems, error) in
+            guard error == nil else {
+                return
+            }
+            self?.updateMediaItems(mediaItems, pageNumber: pageNumber, pageSize: pageSize)
+        }
+    }
+}
+
+private extension WordPressMediaLibraryPickerDataSource {
+    func retrieveMedia(pageNumber: Int, pageSize: Int, completion: @escaping (_ mediaItems: [Media], _ error: Error?) -> Void) {
+        let action = MediaAction.retrieveMediaLibrary(siteID: siteID,
+                                                      pageNumber: pageNumber,
+                                                      pageSize: pageSize) { (mediaItems, error) in
+                                                        guard mediaItems.isEmpty == false else {
+                                                            completion([], error)
+                                                            return
+                                                        }
+                                                        completion(mediaItems, nil)
+        }
+        ServiceLocator.stores.dispatch(action)
+    }
+
+    /// Appends the new media items to the existing media items.
+    ///
+    func updateMediaItems(_ newMediaItems: [Media], pageNumber: Int, pageSize: Int) {
+        let pageFirstIndex = syncingCoordinator.pageFirstIndex
+        let startIndex = (pageNumber - pageFirstIndex) * pageSize
+        let startIndexOfTheNextPage = (pageNumber + 1 - pageFirstIndex) * pageSize
```

Comment:
⚠️  I think renaming these variables could go a long way. I'm not sure I fully understand this method. The way I understand this is that these are the indexes of the "media". Should we rename this to include `"Media"` in the name to disambiguate them from the `pageFirstIndex`? 

Maybe:

- `mediaStartIndex`
- `mediaStartIndexOfTheNextPage`
- `mediaEndIndex`

What do you think? 🙂

Reply:
always happy to rename to improve readability, updated here: https://github.com/woocommerce/woocommerce-ios/pull/2060/commits/11be2271cff0399145f58587496e7c95119b1bbb

Reply:
Love it! Much clearer!